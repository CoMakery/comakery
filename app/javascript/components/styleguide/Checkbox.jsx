import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class Checkbox extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'checkbox',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="checkbox"
          symbolLimit={0}
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default Checkbox
