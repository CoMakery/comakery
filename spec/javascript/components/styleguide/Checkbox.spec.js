import React from 'react'
import { shallow } from 'enzyme'
import Checkbox from 'components/styleguide/Checkbox'

describe('InputFieldUploadFile', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<Checkbox />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.checkbox').props().type).toBe('checkbox')
    expect(wrapper.exists('.input-field--title--counter')).not.toBe()
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<Checkbox className="__test" />)

    expect(wrapper.exists('.checkbox.__test')).toBe(true)
  })
})
