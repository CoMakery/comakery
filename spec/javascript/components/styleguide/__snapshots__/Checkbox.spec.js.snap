// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldUploadFile renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="checkbox"
    disabled={false}
    errorText=""
    eventHandler={[Function]}
    imgInputRef={null}
    imgPreviewDimensions="40x40"
    imgPreviewUrl=""
    imgRequirements=""
    imgVerifier={[Function]}
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    recommended={false}
    required={false}
    selectEntries={Array []}
    symbolLimit={0}
    title="title"
    type="checkbox"
    value=""
  />
</Fragment>
`;
